src/modules/schnorrsig/main_impl.h

```diff
@@ -264,4 +264,206 @@ int secp256k1_schnorrsig_verify(const secp256k1_context* ctx, const unsigned cha
            secp256k1_fe_equal_var(&rx, &r.x);
 }
 
+
+
+/* Initializes SHA256 with fixed midstate. This midstate was computed by applying
+ * SHA256 to SHA256("HalfAgg/randomizer")||SHA256("HalfAgg/randomizer"). */
+void secp256k1_schnorrsig_sha256_tagged_aggregation(secp256k1_sha256 *sha) {
+   secp256k1_sha256_initialize(sha);
+    sha->s[0] = 0xd11f5532ul;
+    sha->s[1] = 0xfa57f70ful;
+    sha->s[2] = 0x5db0d728ul;
+    sha->s[3] = 0xf806ffe1ul;
+    sha->s[4] = 0x1d4db069ul;
+    sha->s[5] = 0xb4d587e1ul;
+    sha->s[6] = 0x50451c2aul;
+    sha->s[7] = 0x10fb63e9ul;
+
+    sha->bytes = 64;
+}
+
+int secp256k1_schnorrsig_inc_aggregate(const secp256k1_context* ctx, unsigned char* aggsig, size_t* aggsig_size, const secp256k1_xonly_pubkey* all_pubkeys, const unsigned char* all_msgs32, const unsigned char* new_sigs64, size_t n_before, size_t n_new) {
+    size_t i;
+    size_t n;
+    int overflow;
+    secp256k1_sha256 hash; 
+    secp256k1_scalar s;
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(aggsig != NULL);
+    ARG_CHECK(aggsig_size != NULL);
+    ARG_CHECK(all_pubkeys != NULL);
+    ARG_CHECK(all_msgs32 != NULL);
+    ARG_CHECK(new_sigs64 != NULL);
+
+    /* Check that aggsig_size is large enough, i.e. aggsig_size >= 32*(n+1) */
+    n = n_before + n_new;
+    if ((*aggsig_size >> 5) <= 0 || ((*aggsig_size >> 5)-1) < n) {
```

Comment:
nit: I think this is a bit more natural to read (but yeah, it's maybe just taste in the end):
```suggestion
    if ((*aggsig_size / 32) == 0 || ((*aggsig_size / 32)-1) < n) {
```